motor/motorid/awsec2/metadata_cmd.go

```diff
@@ -1,55 +1,91 @@
 package awsec2
 
 import (
+	"context"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"strings"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
 	"github.com/cockroachdb/errors"
 	"go.mondoo.com/cnquery/motor/platform"
 	"go.mondoo.com/cnquery/motor/providers/os"
 	"go.mondoo.com/cnquery/motor/providers/os/powershell"
 )
 
 const (
-	identityUrl                   = "http://169.254.169.254/latest/dynamic/instance-identity/document"
-	metadataIdentityScriptWindows = "Invoke-RestMethod -TimeoutSec 1 -URI http://169.254.169.254/latest/dynamic/instance-identity/document -UseBasicParsing | ConvertTo-Json"
+	identityUrl = "http://169.254.169.254/latest/dynamic/instance-identity/document"
+	tagNameUrl  = "http://169.254.169.254/latest/meta-data/tags/instance/Name"
 )
 
-func NewCommandInstanceMetadata(provider os.OperatingSystemProvider, pf *platform.Platform) *CommandInstanceMetadata {
+func NewCommandInstanceMetadata(provider os.OperatingSystemProvider, pf *platform.Platform, config *aws.Config) *CommandInstanceMetadata {
 	return &CommandInstanceMetadata{
 		provider: provider,
 		platform: pf,
+		config:   config,
 	}
 }
 
 type CommandInstanceMetadata struct {
 	provider os.OperatingSystemProvider
 	platform *platform.Platform
+	config   *aws.Config
 }
 
 func (m *CommandInstanceMetadata) Identify() (Identity, error) {
 	instanceDocument, err := m.instanceIdentityDocument()
 	if err != nil {
 		return Identity{}, err
 	}
-
 	// parse into struct
 	doc := imds.InstanceIdentityDocument{}
 	if err := json.NewDecoder(strings.NewReader(instanceDocument)).Decode(&doc); err != nil {
 		return Identity{}, errors.Wrap(err, "failed to decode EC2 instance identity document")
 	}
+
+	name := ""
+	// Note that the tags metadata has to be enabled for this to work. If not, we fallback to trying to get the name
+	// via the aws API (if there's a config provided)
+	taggedName, err := m.instanceTags()
```

Comment:
nit: The function name suggests that multiple tags are returned, but it is the name tag only.